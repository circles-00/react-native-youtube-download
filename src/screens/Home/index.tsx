import Home from './presenter'

export const HOME_SCREEN_KEY = 'Home'

export default Home
