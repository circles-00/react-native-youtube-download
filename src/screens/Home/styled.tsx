// @ts-ignore
import styled from '@emotion/native'
import { StyleSheet } from 'react-native'
import { Colors } from '../../config/Colors'

export const styles = StyleSheet.create({
  mediaMapContainer: {
    backgroundColor: Colors.background
  },
  text: {
    color: '#ffffff'
  },
  searchBar: {
    backgroundColor: '#303030',
    width: '95%',
    marginTop: 25,
    marginLeft: 10,
    height: 60
  },
  searchInput: {
    color: '#a8a4a4',
    fontSize: 19
  }
})
