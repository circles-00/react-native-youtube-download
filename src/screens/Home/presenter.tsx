import React from 'react'
import { View, Text, ScrollView, SafeAreaView } from 'react-native'

import { styles } from './styled'
import GreetingsHeading from './components/GreetingsHeading/GreetingsHeading'
import { Searchbar } from 'react-native-paper'
import { Colors } from '../../constants/Colors'
import MediaCards from './components/MediaCards/MediaCards'
import { MediaMock } from '../../constants/MediaMock'

const Home: React.FC = () => {
  return (
    <SafeAreaView style={styles.mediaMapContainer}>
      <ScrollView>
        <GreetingsHeading title={'Hi There,'} name={'Nikola'} />
        {
          // @ts-ignore
          <Searchbar
            theme={{
              colors: {
                placeholder: styles.searchInput.color,
                text: Colors.primary
              }
            }}
            inputStyle={styles.searchInput}
            style={styles.searchBar}
            placeholder={'Songs, albums or artists'}
          />
        }
        <MediaCards sectionTitle={'Trending Now'} items={MediaMock} />
        <MediaCards sectionTitle={'Popular Now'} items={MediaMock} />
      </ScrollView>
    </SafeAreaView>
  )
}

export default Home
