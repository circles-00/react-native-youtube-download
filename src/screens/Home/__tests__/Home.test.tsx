import React from 'react';
import { render, toJSON } from '@testing-library/react-native';

import Home from '../presenter';

describe('Home - Presenter', () => {
  it('should render', () => {
    const { container } = render(<Home />);
    expect(toJSON(container)).toMatchSnapshot();
  });
});
