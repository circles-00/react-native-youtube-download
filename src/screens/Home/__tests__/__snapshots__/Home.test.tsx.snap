// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home - Presenter should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Open up App.js to start working on your app!
    </Text>
  </View>
</View>
`;
